x/kavamint/types/genesis_test.go

```diff
@@ -1,95 +1,149 @@
 )
 
-func Test_ValidateGenesisAndParams(t *testing.T) {
+func TestGenesis_Default(t *testing.T) {
+	defaultGenesis := types.DefaultGenesisState()
+
+	require.NoError(t, defaultGenesis.Validate())
+
+	defaultParams := types.DefaultParams()
+	assert.Equal(t, defaultParams, defaultGenesis.Params)
+}
+
+func TestGenesis_Validation(t *testing.T) {
 	testCases := []struct {
-		name       string
-		gs         *types.GenesisState
-		shouldPass bool
+		name        string
+		gs          *types.GenesisState
+		expectedErr string
 	}{
 		{
 			"valid - default genesis is valid",
 			types.DefaultGenesisState(),
-			true,
+			"",
 		},
 		{
 			"valid - valid genesis",
 			types.NewGenesisState(
-				types.NewParams(sdk.MustNewDecFromStr("0.1"), sdk.MustNewDecFromStr("0.2")),
+				newValidParams(t),
 				time.Now(),
 			),
-			true,
+			"",
 		},
 		{
 			"valid - no inflation",
 			types.NewGenesisState(
 				types.NewParams(sdk.ZeroDec(), sdk.ZeroDec()),
 				time.Now(),
 			),
-			true,
+			"",
 		},
 		{
-			"invalid - no time set",
+			"vali - no time set",
 			types.NewGenesisState(
-				types.NewParams(sdk.MustNewDecFromStr("0.1"), sdk.MustNewDecFromStr("0.2")),
+				newValidParams(t),
 				time.Time{},
 			),
-			false,
+			"",
 		},
 		{
 			"invalid - community inflation param too big",
 			types.NewGenesisState(
-				// inflation is larger than is allowed!
-				types.NewParams(types.MaxMintingRate.Add(sdk.OneDec()), sdk.ZeroDec()),
+				types.NewParams(types.MaxMintingRate.Add(sdk.NewDecWithPrec(1, 18)), sdk.ZeroDec()),
 				time.Now(),
 			),
-			false,
+			"invalid rate: 0.000000146028999311",
 		},
 		{
 			"invalid - staking reward inflation param too big",
 			types.NewGenesisState(
 				// inflation is larger than is allowed!
-				types.NewParams(sdk.ZeroDec(), types.MaxMintingRate.Add(sdk.OneDec())),
+				types.NewParams(sdk.ZeroDec(), types.MaxMintingRate.Add(sdk.NewDecWithPrec(1, 18))),
 				time.Now(),
 			),
-			false,
+			"invalid rate: 0.000000146028999311",
 		},
 		{
 			"invalid - negative community inflation param",
 			types.NewGenesisState(
-				types.NewParams(sdk.OneDec().MulInt64(-1), sdk.OneDec()),
+				types.NewParams(sdk.OneDec().MulInt64(-1), sdk.ZeroDec()),
 				time.Now(),
 			),
-			false,
+			"invalid rate: -1.000000000000000000",
 		},
 		{
 			"invalid - negative staking inflation param",
 			types.NewGenesisState(
-				types.NewParams(sdk.OneDec(), sdk.OneDec().MulInt64(-1)),
+				types.NewParams(sdk.ZeroDec(), sdk.OneDec().MulInt64(-1)),
 				time.Now(),
 			),
-			false,
+			"invalid rate: -1.000000000000000000",
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			err := tc.gs.Validate()
-			if tc.shouldPass {
-				require.NoError(t, err)
+
+			if tc.expectedErr == "" {
+				assert.Nil(t, err)
 			} else {
-				require.Error(t, err)
+				assert.EqualError(t, err, tc.expectedErr)
 			}
 		})
 	}
 }
+
+func TestGenesis_JSONEncoding(t *testing.T) {
```

Comment:
These json tests help serve as examples and prevent regressions in protobuf encoding/tags